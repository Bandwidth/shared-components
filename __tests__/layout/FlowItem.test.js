import React from 'react';
import toJSON from 'enzyme-to-json';
import {
  FlowItem,
  Label,
  HelpText,
} from '../../src';

describe('the FlowItem component', () => {
  describe('renders with no properties', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowWithTheme(<FlowItem />);
    });

    test('should render', () => {
      expect(wrapper).toHaveLength(1);
    });

    test('renders a label and contents inside a container despite being empty', () => {
      expect(wrapper.children().find(Label)).toHaveLength(1);
      expect(wrapper.children().find(FlowItem.Content)).toHaveLength(1);
      expect(wrapper.children().find(HelpText)).toHaveLength(0);
    });
  });

  describe('renders a label and help text with error prop', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowWithTheme(<FlowItem label="foo" helpText="bar" error />);
    });

    test('renders the correct values', () => {
      const label = wrapper.children().find(Label);
      const helpText = wrapper.children().find(HelpText);
      expect(label).toHaveLength(1);
      expect(helpText).toHaveLength(1);
      expect(label.props().children).toEqual('foo');
      expect(helpText.props().children).toEqual('bar');
      expect(helpText.props().error).toEqual(true);
    });
  });

  describe('renders custom help text', () => {
    let wrapper;
    let customHelpText = <div id="custom">This is custom</div>;

    beforeEach(() => {
      wrapper = shallowWithTheme(<FlowItem helpText={customHelpText} error />);
    });

    test('renders the help text', () => {
      const helpText = wrapper.children().find('#custom');
      expect(helpText).toHaveLength(1);
      expect(helpText.props().error).toEqual(true);
    });
  });

  describe('renders children', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowWithTheme(<FlowItem>foo</FlowItem>);
    });

    test('renders children', () => {
      const container = wrapper.children();
      const content = container.find(FlowItem.Content);
      expect(content).toHaveLength(1);
      expect(content.props().children).toEqual('foo');
    });
  });

  describe('with custom alignment', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowWithTheme(<FlowItem alignment="left">foo</FlowItem>);
    });

    test('applies alignment to container', () => {
      expect(wrapper.props().alignment).toEqual('left');
    });
  });

  describe('can render more content', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowWithTheme(<FlowItem moreContent="foo">bar</FlowItem>);
    });

    test('adds more content', () => {
      const container = wrapper.children();
      const moreContent = container.find(FlowItem.MoreContent);
      expect(moreContent).toHaveLength(1);
      expect(moreContent.props().children).toEqual('foo');
    });
  });

  describe('can render flexible content sizes', () => {
    let wrapper;
    const content = <div style={{ height: '300px' }} />;

    beforeEach(() => {
      wrapper = shallowWithTheme(<FlowItem flexibleContent>{content}</FlowItem>);
    });

    test('renders flexible content', () => {
      const container = wrapper.children();
      const flexContent = container.find(FlowItem.FlexibleContent);
      expect(flexContent).toHaveLength(1);
      expect(flexContent.props().children).toEqual(content);
      expect(container.find(FlowItem.Content)).toHaveLength(0);
    });
  });

  describe('with id and className', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowWithTheme(<FlowItem id="a" className="b" />);
    });

    test('assigns them to the container element', () => {
      expect(wrapper.props().id).toEqual('a');
      expect(wrapper.props().className).toEqual('b');
    });
  });
});

describe('the FlowItem component styling', () => {
  test('with content, label, and helpText', () => {
    const wrapper = mountWithTheme(<FlowItem helpText="foo" label="bar">baz</FlowItem>);
    expect(toJSON(wrapper)).toMatchStyledComponentsSnapshot();
  });
  test('with flexible content', () => {
    const wrapper = mountWithTheme(<FlowItem flexibleContent><div style={{ height: '300px' }} /></FlowItem>);
    expect(toJSON(wrapper)).toMatchStyledComponentsSnapshot();
  });
  test('with error', () => {
    const wrapper = mountWithTheme(<FlowItem error helpText="bar">foo</FlowItem>);
    expect(toJSON(wrapper)).toMatchStyledComponentsSnapshot();
  });
  test('with more content', () => {
    const wrapper = mountWithTheme(<FlowItem moreContent="baz">foo</FlowItem>);
    expect(toJSON(wrapper)).toMatchStyledComponentsSnapshot();
  });
  test('with left align', () => {
    const wrapper = mountWithTheme(<FlowItem alignment="left" helpText="bar" label="baz">foo</FlowItem>);
    expect(toJSON(wrapper)).toMatchStyledComponentsSnapshot();
  });
  test('with right align', () => {
    const wrapper = mountWithTheme(<FlowItem alignment="right" helpText="bar" label="baz">foo</FlowItem>);
    expect(toJSON(wrapper)).toMatchStyledComponentsSnapshot();
  });
});
