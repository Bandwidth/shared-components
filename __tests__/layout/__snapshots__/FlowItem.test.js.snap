// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the FlowItem component styling with content, label, and helpText 1`] = `
.jVAKAe {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.jVAKAe > .Label__Label-jsURsk {
  height: 21px;
  margin: 0;
  margin-bottom: 5.6px;
  overflow-x: hidden;
  -webkit-flex-basis: 21px;
  -ms-flex-basis: 21px;
  flex-basis: 21px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__Content-bZFRMT {
  height: 53px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.jVAKAe > .FlowItem__Content-bZFRMT > * {
  margin: auto;
}

.jVAKAe > .FlowItem__Content-bZFRMT > *:not(:last-child):not(:only-child) {
  margin-right: 30px;
}

.jVAKAe > .FlowItem__FlexibleContent-kAxlQq {
  height: auto;
  -webkit-flex-basis: 53px;
  -ms-flex-basis: 53px;
  flex-basis: 53px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.jVAKAe > .HelpText__HelpText-cZQOvv {
  min-height: 18px;
  margin: 0;
  margin-top: 5.6px;
  -webkit-flex-basis: 18px;
  -ms-flex-basis: 18px;
  flex-basis: 18px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__MoreContent-eNWnOb {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.gdcnMF {
  font-size: 1em;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  font-weight: 600;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
  padding-bottom: 0.4em;
  color: #666;
  background: transparent;
  opacity: 1;
  display: block;
  line-height: 1.5em;
}

.lhMiJf {
  color: #7f8897;
  font-style: italic;
  font-weight: 300;
  padding: 0.4em 0 0 0;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
}

<FlowItem
  __styled-components__={[Function]}
  alignment="stretch"
  className={null}
  error={false}
  flexibleContent={false}
  helpText="foo"
  id={null}
  label="bar"
  moreContent={null}
  suppressLabel={false}
>
  <FlowItem__Container
    alignment="stretch"
    className={null}
    id={null}
  >
    <div
      className="FlowItem__Container-krUYur jVAKAe"
      id={null}
    >
      <Label__Label
        className={null}
        disabled={false}
        id={null}
      >
        <label
          className="Label__Label-jsURsk gdcnMF"
          disabled={false}
          id={null}
        >
          bar
        </label>
      </Label__Label>
      <FlowItem__Content>
        <div
          className="FlowItem__Content-bZFRMT cwqVhq"
        >
          baz
        </div>
      </FlowItem__Content>
      <HelpText__HelpText
        className={null}
        error={false}
        id={null}
      >
        <div
          className="HelpText__HelpText-cZQOvv lhMiJf"
          id={null}
        >
          foo
        </div>
      </HelpText__HelpText>
    </div>
  </FlowItem__Container>
</FlowItem>
`;

exports[`the FlowItem component styling with error 1`] = `
.jVAKAe {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.jVAKAe > .Label__Label-jsURsk {
  height: 21px;
  margin: 0;
  margin-bottom: 5.6px;
  overflow-x: hidden;
  -webkit-flex-basis: 21px;
  -ms-flex-basis: 21px;
  flex-basis: 21px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__Content-bZFRMT {
  height: 53px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.jVAKAe > .FlowItem__Content-bZFRMT > * {
  margin: auto;
}

.jVAKAe > .FlowItem__Content-bZFRMT > *:not(:last-child):not(:only-child) {
  margin-right: 30px;
}

.jVAKAe > .FlowItem__FlexibleContent-kAxlQq {
  height: auto;
  -webkit-flex-basis: 53px;
  -ms-flex-basis: 53px;
  flex-basis: 53px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.jVAKAe > .HelpText__HelpText-cZQOvv {
  min-height: 18px;
  margin: 0;
  margin-top: 5.6px;
  -webkit-flex-basis: 18px;
  -ms-flex-basis: 18px;
  flex-basis: 18px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__MoreContent-eNWnOb {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.gdcnMF {
  font-size: 1em;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  font-weight: 600;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
  padding-bottom: 0.4em;
  color: #666;
  background: transparent;
  opacity: 1;
  display: block;
  line-height: 1.5em;
}

.irGmEk {
  color: #e8562e;
  font-style: italic;
  font-weight: 300;
  padding: 0.4em 0 0 0;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
}

<FlowItem
  __styled-components__={[Function]}
  alignment="stretch"
  className={null}
  error={true}
  flexibleContent={false}
  helpText="bar"
  id={null}
  label={null}
  moreContent={null}
  suppressLabel={false}
>
  <FlowItem__Container
    alignment="stretch"
    className={null}
    id={null}
  >
    <div
      className="FlowItem__Container-krUYur jVAKAe"
      id={null}
    >
      <Label__Label
        className={null}
        disabled={false}
        id={null}
      >
        <label
          className="Label__Label-jsURsk gdcnMF"
          disabled={false}
          id={null}
        />
      </Label__Label>
      <FlowItem__Content>
        <div
          className="FlowItem__Content-bZFRMT cwqVhq"
        >
          foo
        </div>
      </FlowItem__Content>
      <HelpText__HelpText
        className={null}
        error={true}
        id={null}
      >
        <div
          className="HelpText__HelpText-cZQOvv irGmEk"
          id={null}
        >
          bar
        </div>
      </HelpText__HelpText>
    </div>
  </FlowItem__Container>
</FlowItem>
`;

exports[`the FlowItem component styling with flexible content 1`] = `
.jVAKAe {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.jVAKAe > .Label__Label-jsURsk {
  height: 21px;
  margin: 0;
  margin-bottom: 5.6px;
  overflow-x: hidden;
  -webkit-flex-basis: 21px;
  -ms-flex-basis: 21px;
  flex-basis: 21px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__Content-bZFRMT {
  height: 53px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.jVAKAe > .FlowItem__Content-bZFRMT > * {
  margin: auto;
}

.jVAKAe > .FlowItem__Content-bZFRMT > *:not(:last-child):not(:only-child) {
  margin-right: 30px;
}

.jVAKAe > .FlowItem__FlexibleContent-kAxlQq {
  height: auto;
  -webkit-flex-basis: 53px;
  -ms-flex-basis: 53px;
  flex-basis: 53px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.jVAKAe > .HelpText__HelpText-cZQOvv {
  min-height: 18px;
  margin: 0;
  margin-top: 5.6px;
  -webkit-flex-basis: 18px;
  -ms-flex-basis: 18px;
  flex-basis: 18px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__MoreContent-eNWnOb {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.gdcnMF {
  font-size: 1em;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  font-weight: 600;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
  padding-bottom: 0.4em;
  color: #666;
  background: transparent;
  opacity: 1;
  display: block;
  line-height: 1.5em;
}

<FlowItem
  __styled-components__={[Function]}
  alignment="stretch"
  className={null}
  error={false}
  flexibleContent={true}
  helpText={null}
  id={null}
  label={null}
  moreContent={null}
  suppressLabel={false}
>
  <FlowItem__Container
    alignment="stretch"
    className={null}
    id={null}
  >
    <div
      className="FlowItem__Container-krUYur jVAKAe"
      id={null}
    >
      <Label__Label
        className={null}
        disabled={false}
        id={null}
      >
        <label
          className="Label__Label-jsURsk gdcnMF"
          disabled={false}
          id={null}
        />
      </Label__Label>
      <FlowItem__FlexibleContent>
        <div
          className="FlowItem__FlexibleContent-kAxlQq hFXbsh"
        >
          <div
            style={
              Object {
                "height": "300px",
              }
            }
          />
        </div>
      </FlowItem__FlexibleContent>
    </div>
  </FlowItem__Container>
</FlowItem>
`;

exports[`the FlowItem component styling with left align 1`] = `
.uxbnB {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.uxbnB > .Label__Label-jsURsk {
  height: 21px;
  margin: 0;
  margin-bottom: 5.6px;
  overflow-x: hidden;
  -webkit-flex-basis: 21px;
  -ms-flex-basis: 21px;
  flex-basis: 21px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.uxbnB > .FlowItem__Content-bZFRMT {
  height: 53px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.uxbnB > .FlowItem__Content-bZFRMT > * {
  margin: auto;
}

.uxbnB > .FlowItem__Content-bZFRMT > *:not(:last-child):not(:only-child) {
  margin-right: 30px;
}

.uxbnB > .FlowItem__FlexibleContent-kAxlQq {
  height: auto;
  -webkit-flex-basis: 53px;
  -ms-flex-basis: 53px;
  flex-basis: 53px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.uxbnB > .HelpText__HelpText-cZQOvv {
  min-height: 18px;
  margin: 0;
  margin-top: 5.6px;
  -webkit-flex-basis: 18px;
  -ms-flex-basis: 18px;
  flex-basis: 18px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.uxbnB > .FlowItem__MoreContent-eNWnOb {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.gdcnMF {
  font-size: 1em;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  font-weight: 600;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
  padding-bottom: 0.4em;
  color: #666;
  background: transparent;
  opacity: 1;
  display: block;
  line-height: 1.5em;
}

.lhMiJf {
  color: #7f8897;
  font-style: italic;
  font-weight: 300;
  padding: 0.4em 0 0 0;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
}

<FlowItem
  __styled-components__={[Function]}
  alignment="left"
  className={null}
  error={false}
  flexibleContent={false}
  helpText="bar"
  id={null}
  label="baz"
  moreContent={null}
  suppressLabel={false}
>
  <FlowItem__Container
    alignment="left"
    className={null}
    id={null}
  >
    <div
      className="FlowItem__Container-krUYur uxbnB"
      id={null}
    >
      <Label__Label
        className={null}
        disabled={false}
        id={null}
      >
        <label
          className="Label__Label-jsURsk gdcnMF"
          disabled={false}
          id={null}
        >
          baz
        </label>
      </Label__Label>
      <FlowItem__Content>
        <div
          className="FlowItem__Content-bZFRMT cwqVhq"
        >
          foo
        </div>
      </FlowItem__Content>
      <HelpText__HelpText
        className={null}
        error={false}
        id={null}
      >
        <div
          className="HelpText__HelpText-cZQOvv lhMiJf"
          id={null}
        >
          bar
        </div>
      </HelpText__HelpText>
    </div>
  </FlowItem__Container>
</FlowItem>
`;

exports[`the FlowItem component styling with more content 1`] = `
.jVAKAe {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.jVAKAe > .Label__Label-jsURsk {
  height: 21px;
  margin: 0;
  margin-bottom: 5.6px;
  overflow-x: hidden;
  -webkit-flex-basis: 21px;
  -ms-flex-basis: 21px;
  flex-basis: 21px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__Content-bZFRMT {
  height: 53px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.jVAKAe > .FlowItem__Content-bZFRMT > * {
  margin: auto;
}

.jVAKAe > .FlowItem__Content-bZFRMT > *:not(:last-child):not(:only-child) {
  margin-right: 30px;
}

.jVAKAe > .FlowItem__FlexibleContent-kAxlQq {
  height: auto;
  -webkit-flex-basis: 53px;
  -ms-flex-basis: 53px;
  flex-basis: 53px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.jVAKAe > .HelpText__HelpText-cZQOvv {
  min-height: 18px;
  margin: 0;
  margin-top: 5.6px;
  -webkit-flex-basis: 18px;
  -ms-flex-basis: 18px;
  flex-basis: 18px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__MoreContent-eNWnOb {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.gdcnMF {
  font-size: 1em;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  font-weight: 600;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
  padding-bottom: 0.4em;
  color: #666;
  background: transparent;
  opacity: 1;
  display: block;
  line-height: 1.5em;
}

<FlowItem
  __styled-components__={[Function]}
  alignment="stretch"
  className={null}
  error={false}
  flexibleContent={false}
  helpText={null}
  id={null}
  label={null}
  moreContent="baz"
  suppressLabel={false}
>
  <FlowItem__Container
    alignment="stretch"
    className={null}
    id={null}
  >
    <div
      className="FlowItem__Container-krUYur jVAKAe"
      id={null}
    >
      <Label__Label
        className={null}
        disabled={false}
        id={null}
      >
        <label
          className="Label__Label-jsURsk gdcnMF"
          disabled={false}
          id={null}
        />
      </Label__Label>
      <FlowItem__Content>
        <div
          className="FlowItem__Content-bZFRMT cwqVhq"
        >
          foo
        </div>
      </FlowItem__Content>
      <FlowItem__MoreContent>
        <div
          className="FlowItem__MoreContent-eNWnOb hbqMVH"
        >
          baz
        </div>
      </FlowItem__MoreContent>
    </div>
  </FlowItem__Container>
</FlowItem>
`;

exports[`the FlowItem component styling with right align 1`] = `
.dYdFrk {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.dYdFrk > .Label__Label-jsURsk {
  height: 21px;
  margin: 0;
  margin-bottom: 5.6px;
  overflow-x: hidden;
  -webkit-flex-basis: 21px;
  -ms-flex-basis: 21px;
  flex-basis: 21px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.dYdFrk > .FlowItem__Content-bZFRMT {
  height: 53px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.dYdFrk > .FlowItem__Content-bZFRMT > * {
  margin: auto;
}

.dYdFrk > .FlowItem__Content-bZFRMT > *:not(:last-child):not(:only-child) {
  margin-right: 30px;
}

.dYdFrk > .FlowItem__FlexibleContent-kAxlQq {
  height: auto;
  -webkit-flex-basis: 53px;
  -ms-flex-basis: 53px;
  flex-basis: 53px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.dYdFrk > .HelpText__HelpText-cZQOvv {
  min-height: 18px;
  margin: 0;
  margin-top: 5.6px;
  -webkit-flex-basis: 18px;
  -ms-flex-basis: 18px;
  flex-basis: 18px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.dYdFrk > .FlowItem__MoreContent-eNWnOb {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.gdcnMF {
  font-size: 1em;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  font-weight: 600;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
  padding-bottom: 0.4em;
  color: #666;
  background: transparent;
  opacity: 1;
  display: block;
  line-height: 1.5em;
}

.lhMiJf {
  color: #7f8897;
  font-style: italic;
  font-weight: 300;
  padding: 0.4em 0 0 0;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
}

<FlowItem
  __styled-components__={[Function]}
  alignment="right"
  className={null}
  error={false}
  flexibleContent={false}
  helpText="bar"
  id={null}
  label="baz"
  moreContent={null}
  suppressLabel={false}
>
  <FlowItem__Container
    alignment="right"
    className={null}
    id={null}
  >
    <div
      className="FlowItem__Container-krUYur dYdFrk"
      id={null}
    >
      <Label__Label
        className={null}
        disabled={false}
        id={null}
      >
        <label
          className="Label__Label-jsURsk gdcnMF"
          disabled={false}
          id={null}
        >
          baz
        </label>
      </Label__Label>
      <FlowItem__Content>
        <div
          className="FlowItem__Content-bZFRMT cwqVhq"
        >
          foo
        </div>
      </FlowItem__Content>
      <HelpText__HelpText
        className={null}
        error={false}
        id={null}
      >
        <div
          className="HelpText__HelpText-cZQOvv lhMiJf"
          id={null}
        >
          bar
        </div>
      </HelpText__HelpText>
    </div>
  </FlowItem__Container>
</FlowItem>
`;

exports[`the FlowItem component styling with suppressed label 1`] = `
.jVAKAe {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.jVAKAe > .Label__Label-jsURsk {
  height: 21px;
  margin: 0;
  margin-bottom: 5.6px;
  overflow-x: hidden;
  -webkit-flex-basis: 21px;
  -ms-flex-basis: 21px;
  flex-basis: 21px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__Content-bZFRMT {
  height: 53px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.jVAKAe > .FlowItem__Content-bZFRMT > * {
  margin: auto;
}

.jVAKAe > .FlowItem__Content-bZFRMT > *:not(:last-child):not(:only-child) {
  margin-right: 30px;
}

.jVAKAe > .FlowItem__FlexibleContent-kAxlQq {
  height: auto;
  -webkit-flex-basis: 53px;
  -ms-flex-basis: 53px;
  flex-basis: 53px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.jVAKAe > .HelpText__HelpText-cZQOvv {
  min-height: 18px;
  margin: 0;
  margin-top: 5.6px;
  -webkit-flex-basis: 18px;
  -ms-flex-basis: 18px;
  flex-basis: 18px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__MoreContent-eNWnOb {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

<FlowItem
  __styled-components__={[Function]}
  alignment="stretch"
  className={null}
  error={false}
  flexibleContent={false}
  helpText={null}
  id={null}
  label={null}
  moreContent={null}
  suppressLabel={true}
>
  <FlowItem__Container
    alignment="stretch"
    className={null}
    id={null}
  >
    <div
      className="FlowItem__Container-krUYur jVAKAe"
      id={null}
    >
      <FlowItem__Content>
        <div
          className="FlowItem__Content-bZFRMT cwqVhq"
        >
          foo
        </div>
      </FlowItem__Content>
    </div>
  </FlowItem__Container>
</FlowItem>
`;
